.github/workflows/e2e-workflow.yaml

```diff
@@ -0,0 +1,38 @@
+name: E2E Workflow
+
+on:
+  # Run this on pushes to main
+  push:
+    branches:
+    - main
+
+  # Or when PR operations are done
+  pull_request:
```

Comment:
do we want to add `branches: [ master ]` so it runs only on PRs to master?

Reply:
`master` branch will soon be renamed to `main` so I think this is fine as is

Reply:
then the same question with `main` :)

in `main.yml` we have this guard so PRs to other branches do not run. i think that we should be aligned between GH actions, unless we have a good reason not to be. I'm ok with not having this guard but let's be consistent across actions so let's remove it from there.